### PR TITLE
Allow custom templates to be provided separately for each template

### PR DESCRIPTION
**Pull request by ANorwell**
This is a proposed fix for to https://github.com/sbt/sbt-native-packager/issues/937.

Unfortunately, this is a breaking change to the templates directory layout, but it seems to be necessary to allow the different templates to be set independently.

Also, I'm not familiar with the sbt-native-packager codebase. This seems to be the right change, but it's possible this will have unintended consequences.

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/packager/archetypes/systemloader/package.scala [hunk 1]
```diff
@@ -71,5 +71,5 @@ package object systemloader {
     loader.toString + "/" + name
 
   private def overrideFromFile(sourceDirectory: File, loader: ServerLoader, name: String): Option[URL] =
-    Option(sourceDirectory / "templates" / "systemloader" / loader.toString).filter(_.exists).map(_.toURI.toURL)
+    Option(sourceDirectory / "templates" / "systemloader" / loader.toString / name).filter(_.exists).map(_.toURI.toURL)
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/archetypes/systemloader/package.scala]: Thanks for fixing this. This quite some bug. Along with 6 tests that tested the wrong implementation:

```
[error] (*:scripted) Failed tests:
[error] 	rpm / override-start-script-systemd
[error] 	rpm / override-start-script-systemv
[error] 	rpm / override-start-script-upstart
[error] 	debian / override-start-script-systemd
[error] 	debian / override-start-script-systemv
[error] 	debian / override-start-script-upstart
```

**Comment by typesafehub-validator** [comment]
Hi @ANorwell,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by ANorwell** [comment]
I fixed the tests, and also added two new tests for overriding loader-scripts, one for deb and one for rpm.

File: src/sbt-test/debian/override-loader-functions/build.sbt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  with multiple lines."""
+
+TaskKey[Unit]("check-loader-functions") <<= (target, streams) map { (target, out) =>
+  val extracted = target / "tmp" / "extracted-package"
+  extracted.mkdirs()
+  Seq("dpkg-deb", "-e", (target / "debian-test_0.1.0_all.deb").absolutePath, extracted.absolutePath).!
+
+  val script = IO.read(extracted / "postinst")
+  
+  assert(script.contains("# right systemd template"), s"override script wasn't picked, script is\n$script")
```
Comment (muuki88) [inline src/sbt-test/debian/override-loader-functions/build.sbt]: This checks that the `loader-functions` has be correctly inserted. I think this check is enough, even though I wouldn't my checking that the `wrong systemd template` isn't contained in the script.

Comment (ANorwell) [inline src/sbt-test/debian/override-loader-functions/build.sbt]: added such an assertion

**Comment by muuki88** [comment]
Thanks again for your time and work 😊
I'll release this ASAP

**Pull request merged**